### DISCUSSION TITLE
[Greenfield API] Add 'comment' parameter to create store on-chain wallet transaction endpoint

### DISCUSSION THREAD

**Discussion by danielalexiuc** [btcpayserver/btcpayserver#3527] [Ideas / Feature Requests]
BTCPayServer already has a 'comment' field associated with an on-chain wallet transaction - that can be set in the UI, and retrieved by the API. However it cannot currently be _SET_ using the API.

I propose adding the comment field as a new parameter in this POST endpoint: 
`/api/v1/stores/{storeId}/payment-methods/OnChain/{cryptoCode}/wallet/transactions`

My use case: currently if you POST a new transaction to the API, but get no response (e.g. timeout), you do not have any unique identifier (e.g. transaction id) with which you could later look up and find the payment attempt. Setting something into the comment field will allow you locate the transaction easily.

**Comment by bolatovumar** [comment]
@danielalexiuc I started a PR to add the ability to add comment and labels to the transaction through the API here: https://github.com/btcpayserver/btcpayserver/pull/3561

This PR adds support for doing a PATCH on a transaction after it was created, now while it's being created though. Once that PR is merged though, the code could be reused for doing what you are requesting here.
